JavaFXUserInterface/src/main/java/controllers/AssetTypeInfoController.java

```diff
@@ -249,14 +264,33 @@ private void attachEventsModelTab() {
                 trainSlider.setMax(nbOfAssets - testSlider.getValue());
                 testSize = (int) testSlider.getValue();
             });
+
+
             try {
-                evaluateAllModelsBtn.setOnMouseClicked(mouseEvent -> evaluateAllModels());
+                evaluateAllModelsBtn.setOnMouseClicked(mouseEvent -> {
+                    for (Model model : modelObservableList){
+                        String modelName = model.getModelName();
+                        int assetTypeID = Integer.parseInt(assetType.getId());
+                        int trainAssets = (int) trainSlider.getValue() + 1;
+                        int testAssets = (int) trainSlider.getValue() + 1 + (int) testSlider.getValue();
+                        try {
+                            ModelStrategy modelStrategy = modelDAO.getModelStrategy(associatedModelID,assetTypeID);
+                            modelStrategy.setTrainAssets(trainAssets);
+                            modelStrategy.setTestAssets(testAssets);
+                            modelDAO.updateModelStrategy(modelStrategy,associatedModelID,assetTypeID);
+                        } catch (SQLException throwables) {
+                            throwables.printStackTrace();
+                        }
+                         }
+
+                });
```

Comment:
This code is duplicated almost line by line on line 510, please reduce the duplication

Reply:
moved duplicated code to separate function